stumpy/stump.py

```diff
@@ -443,6 +466,8 @@ def stump(T_A, m, T_B=None, ignore_trivial=True):
     M_T, Σ_T = core.compute_mean_std(T_A, m)
     μ_Q, σ_Q = core.compute_mean_std(T_B, m)
 
+    T_A[np.isnan(T_A)] = 0
```

Comment:
Remind me, what is the logic here again?

Reply:
We want to have the mean for every sub sequence that contains a inf/nan value set to inf, because this is how unwanted sequences are being filtered out in the end. But our time series T_A is not allowed to have nans anywhere, because of how QT is computed (incrementally). If it were to have nans, QT would be nan for all values after a certain point. Thats why we set T_A to zero, to allow computations, and the artificially change only the unwanted sequences back to nan using the mean. That is actually exactly the job of the inf_mask.

Reply:
Perhaps this comment should be added here? 

`  # Remove all nan values from T_B only after first profile is calculated`

Reply:
I missed this comment that helps explain the logic as well. https://github.com/TDAmeritrade/stumpy/pull/122#issue-364812444